### PR TITLE
[timeseries] Fixed metrics calculation. Added GluonTS for baselines. Improved AutoGluonTS implementation.

### PR DESCRIPTION
**Pull request by limpbot**
Hi @sebhrusen, @PGijsbers,
In this PR there are the following changes, I would highly appreciate it if you could review them.

Renaming forecast_range_in_steps to forecast_horizon_in_steps, because it is much more popular in the research field.

Fixed metrics calculation. A) Renamed y_past_period_error to naive_1_error, because we do not know periodicity/seasonality. B) Renamed ncrps (Normalized Continuous Ranked Probability Score) to mwql (Mean Weighted Quantile Loss), because that is what is actually calculated. This is an approximation of ncrps. C) Added item_id column to results, to be able to calculate itemwise_mean. D) Introduced finite_mean to calculate correct metrics, even if there are some NaNs.

Added GluonTS framework, because it allows to calculate multiple timeseries baselines. GluonTS_* 1) Prophet, 2) DeepAR, 3) NBEATS, 4) NPTS, 5) SeasonalNaive, 6) SimpleFeedForward, 7) MQCNN, 8) MQRNN, 9) TFT, 10) ARIMA, 11) ETS, 12) STL-AR, 13) Theta.

Added improvements for AutoGluonTS. A) Do not deepcopy the dataset to save memory. B) Use correct module to retrieve the AutoGluon version.

FYI: @Innixma @canerturkmen @gidler

### CODE REVIEW

**Review by PGijsbers — COMMENTED** [review]
Sorry, I don't have time for a closer look right now, but some things did jump out at me already. Thanks for the continued work on this feature :)

File: frameworks/GluonTS/exec.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    elif model == "STL-AR":
+        predictor = RForecastPredictor(prediction_length=prediction_length, freq=freq, method_name="stlar")
+
+    elif model == "Theta":
+        predictor = RForecastPredictor(prediction_length=prediction_length, freq=freq, method_name="thetaf")
+
+    else:
+        msg = f"Not implemented model {model}."
+        raise ValueError(msg)
+    return predictor
```
[the remark this answers is not in the document]
Comment (Innixma) [reply frameworks/GluonTS/exec.py:163]: GluonTS is not an AutoML framework. You can think of these as baseline algorithms implemented in a model zoo (GluonTS being the model zoo).

This is similar to using AutoGluon to get baselines for individual models like `XGBoost`, `CatBoost`, etc.

File: resources/frameworks.yaml [hunk 2]
```diff
@@ -203,6 +203,106 @@ AutoGluonTS:
   setup_env:
     MODULE: timeseries
 
+GluonTS:
```
Comment (Innixma) [inline resources/frameworks.yaml:206]: Add comment in this section that these frameworks are all baselines and not AutoML systems.

File: frameworks/GluonTS/exec.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (PGijsbers) [reply frameworks/GluonTS/exec.py:163]: Thanks for the clarification!

File: frameworks/GluonTS/exec.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    def on_epoch_end(
+        self,
+        **kwargs,
+    ) -> bool:
+        if self.time_limit is not None:
+            cur_time = time.time()
+            if cur_time - self.start_time > self.time_limit:
+                log.warning("Time limit exceed during training, stop training.")
+                return False
+        return True
```
[the remark this answers is not in the document]
Comment (PGijsbers) [reply frameworks/GluonTS/exec.py:185]: Resolving this as I misunderstood the situation.

**Pull request closed without merging**